### ISSUE TITLE
Hadoop Parquet reader fails to open file, illegal character in schema

### ISSUE THREAD

**Issue by plaisted** [aloneguid/parquet-dotnet#93]
**Version:** Parquet.Net 3.8.3

**Runtime Version:** .net 5

**OS:** Windows

#### Expected behavior

Parquet files created by Parquet.Net load using hadoop Parquet reader.

#### Actual behavior

Exception occurs:

```java
org.apache.avro.SchemaParseException: Illegal character in: parquet-dotnet-schema
        at org.apache.avro.Schema.validateName(Schema.java:1068)
        at org.apache.avro.Schema.access$200(Schema.java:79)
        at org.apache.avro.Schema$Name.<init>(Schema.java:439)
        at org.apache.avro.Schema.createRecord(Schema.java:145) 
        at org.apache.parquet.avro.AvroSchemaConverter.convertFields(AvroSchemaConverter.java:225)
        at org.apache.parquet.avro.AvroSchemaConverter.convert(AvroSchemaConverter.java:209)
        at org.apache.parquet.avro.AvroReadSupport.prepareForRead(AvroReadSupport.java:124)
        at org.apache.parquet.hadoop.InternalParquetRecordReader.initialize(InternalParquetRecordReader.java:179)
        at org.apache.parquet.hadoop.ParquetReader.initReader(ParquetReader.java:152)
        at org.apache.parquet.hadoop.ParquetReader.read(ParquetReader.java:128)
```

This appears to be an issue with the '-' characters in the auto-generated Thrift footer by Parquet.Net. Note if I go in and manually edit this column name (eg. replace '-' with '_' in the parquet binary) it loads properly. I don't see any way to prevent this from being added to the files.

**Comment by aloneguid** [comment]
Is this an issue with Hadoop?

**Comment by plaisted** [comment]
I'm not sure, I'm not very familiar with Thrift.  It seems the Hadoop Parquet reader is validating the names against the Avro spec for names (https://avro.apache.org/docs/current/spec.html#names) which seems odd to me, maybe it's storing it internally as Avro. Either way it would be great to be able to override the footer name to make it compatible. Right now I'm having to replace the value afterwards which is a pain with binary files.

**Comment by aloneguid** [comment]
oh I see what you mean

**Comment by aloneguid** [comment]
@plaisted should be fixed in v3.8.4, please check ;)

**Comment by plaisted** [comment]
Just reran files without any modifications afterwards and they imported error free. Thank you for quick response!
